cmd/kured/main.go

```diff
@@ -95,11 +103,14 @@ func init() {
 
 func main() {
 	rootCmd := &cobra.Command{
-		Use:    "kured",
-		Short:  "Kubernetes Reboot Daemon",
-		PreRun: flagCheck,
-		Run:    root}
-
+		Use:               "kured",
+		Short:             "Kubernetes Reboot Daemon",
+		PreRun:            flagCheck,
+		PersistentPreRunE: bindViper,
```

Comment:
@evrardjp addressed the first comment (switched the order)

about Viper/Cobra binding - by far not the expert, I was mostly following the blog here https://carolynvanslyck.com/blog/2020/08/sting-of-the-viper/ and relevant Viper docs (https://github.com/spf13/viper#working-with-environment-variables)

1. To my understanding, that is the intended method, at this point it binds to the define cobra commands and just performs the environment variable binding if relevant

2. bindViper defines the prefix of env variables to consider `KURED_*` and `AutomaticEnv` does the matching between flags & env variables, with flags taking precedence and respecting their defaults, same validation for the values is applied as currently used for the flag values (for example if you use `KURED_LOCK_TTL=10` <-- that won't work, since it's not a `Time.duration`, the `bindFlags` function handles normalizing the matching of env variables to flags by replacing `-` with `_` - this just augments `AutomaticEnv` to handle the flags that have `-` in them that `AutomaticEnv` doesn't handle on it's own.

Reply:
As for testing, I tried looking into adding a unit test, however since Viper will only populate the flags from environment variable when the command is executed I don't see a (straightforward) way to make it work, since it fails to run due to missing serviceaccount token file.

perhaps this can be added to the kind tests, but I'm a bit out of my depth there...

OTOH, since KURED_NODE_ID now gets populated using this new code, this is actually being tested already :-)